app/src/main/java/ch/hikemate/app/ui/saved/SavedHikesScreen.kt

```diff
@@ -62,23 +60,27 @@ object SavedHikesScreen {
 }
 
 @Composable
-fun SavedHikesScreen(
-    savedHikesViewModel: SavedHikesViewModel = viewModel(factory = SavedHikesViewModel.Factory),
-    navigationActions: NavigationActions
-) {
+fun SavedHikesScreen(hikesViewModel: HikesViewModel, navigationActions: NavigationActions) {
   BottomBarNavigation(
       onTabSelect = { route -> navigationActions.navigateTo(route) },
       tabList = LIST_TOP_LEVEL_DESTINATIONS,
       selectedItem = Route.SAVED_HIKES,
   ) { paddingValues ->
     var currentSection by remember { mutableStateOf(SavedHikesSection.Saved) }
-    val loading by savedHikesViewModel.loadingSavedHikes.collectAsState()
-    val errorMessageId by savedHikesViewModel.errorMessageId.collectAsState()
-    val savedHikes by savedHikesViewModel.savedHike.collectAsState()
+    val loading by hikesViewModel.loading.collectAsState()
+    val savedHikes by hikesViewModel.hikeFlows.collectAsState()
+    val selectedHike by hikesViewModel.selectedHike.collectAsState()
+    val osmDataAvailable by hikesViewModel.allOsmDataLoaded.collectAsState()
```

Comment:
Question: Why don't you use the `hikesType` here ?

Reply:
No excuse your honor, I was only using `loading`, but if there's an error it would be better to use `hikesType`